sycl/doc/OptionalDeviceFeatures.md

```diff
@@ -629,11 +629,122 @@ example, if a new device is released before there is a new DPC++ release.  In
 fact, the DPC++ driver supports a command line option which allows the user
 to select an alternate configuration file.
 
-**TODO**: More information will be inserted here when we merge
-[this separate PR][7] into this design document.
+**TODO**: 
+* The names of the devices in the configuration file.
+* Location of the default device configuration file
+
+#### New features in clang compilation driver and tools
+
+NOTE: the term *device binary image* or *image* is used to refer to a device
+code form consumable by the SYCL runtime. Earlier device code forms are referred
+to as *device code module* or *device IR module*. In case of AOT, device binary
+image is a natively compiled binary, and IR module - either a SPIR-V or LLVMIR
+bitcode module.
+
+##### Overview
+After the `sycl-post-link` performs necessary aspect usage analysis and splits
+the incoming monolythic device code module into pieces - smaller device code
+modules - it outputs a file table as a result. Each row in the table corresponds
+to an individual output module, and each element of a row is a name of a file
+containing necessary information about the module, such as the code itself, its
+properties.
+
+At the action graph building stage for each requested AOT compilation target -
+SPIR-V-based (such as Gen targets) and/or non-SPIR-V-based (such as PTX) - the
+driver adds an `aspect-filter` action which filters out input file table rows
+with device code modules using features unsupported on current target. Then the
+output table goes as input into the AOT stage, and the prior filtering
+guarantees that the AOT compiler will not encounter device code it can't
+compile. In the extreme case when all device code
+modules use unsupported aspects, the input file table will be empty. The picture
+below illustrates the action graph built by the clang driver along with file
+lists and tables generated and consumed by various nodes of the graph. The
+example set of targets used for the illustration is 4 targets
+- spir64 (runtime JITted SPIR-V)
+- AOT targets
+    - non-SPIR-V based
+        - nvptx (PTX)
+    - SPIR-V based
+        - gen_11 (Intel Gen)
+        - x86_64_avx512 (AVX512)
```

Comment:
It would be good to align the target names here and the names in the SVG file with the names you propose below in "AOT target identification".  For example, there is no target named "nvptx", so this should probably list "ptx64" instead.  Also, there is no entry "gen_11" in the table below.  Should that be added to the table?  Or, maybe this example should use "intel_gpu_9_0" instead?  Note that these names also show up in several places in the SVG diagram, so those should be updated too.

Also, update the example configuration file above in the section titled "Device configuration file" to use device names that you propose.

Reply:
right, good catch